clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -529,10 +529,9 @@ public ConsumerGroupHeartbeatRequestData buildRequestData() {
             // MemberEpoch - always sent
             data.setMemberEpoch(membershipManager.memberEpoch());
 
-            // InstanceId - only sent if has changed since the last heartbeat
-            // Always send when leaving the group as a static member
+            // InstanceId - send when leaving the group as a static member
             membershipManager.groupInstanceId().ifPresent(groupInstanceId -> {
```

Comment:
The field is at minimum also required in the first request too. Overall, I think that we should just set it all the time if `groupInstanceId` is defined. This is actually was the server expects in the current implementation. We may be able to relax this requirement in the future though.

Reply:
Agree, the current coordinator expectation seems to me like the right thing to do, and we should send the `groupInstanceId` all the time (in the same way that we send the `memberId` all the time).